Apps/Sandcastle/gallery/Clustering.html

```diff
@@ -0,0 +1,158 @@
+        require.config({
+            baseUrl : '../../../Source',
+            waitSeconds : 60
+        });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+    #toolbar {
+        background: rgba(42, 42, 42, 0.8);
+        padding: 4px;
+        border-radius: 4px;
+    }
+    #toolbar input {
+        vertical-align: middle;
+        padding-top: 2px;
+        padding-bottom: 2px;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar">
+    <table>
+        <tbody><tr>
+            <td>pixel range</td>
+            <td>
+                <input type="range" min="1" max="200" step="1" data-bind="value: pixelRange, valueUpdate: 'input'">
+                <input type="text" size="2" data-bind="value: pixelRange">
+            </td>
+        </tr>
+        <tr>
+            <td>minimum cluster size</td>
+            <td>
+                <input type="range" min="2" max="20" step="1" data-bind="value: minimumClusterSize, valueUpdate: 'input'">
+                <input type="text" size="2" data-bind="value: minimumClusterSize">
+            </td>
+        </tr>
+        </tbody>
+    </table>
+    <div id="toggleClustering"></div>
+    <div id="customStyling"></div>
+</div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer');
+
+var options = {
+    camera : viewer.scene.camera,
+    canvas : viewer.scene.canvas
+};
+var dataSourcePromise = viewer.dataSources.add(Cesium.KmlDataSource.load('../../SampleData/kml/facilities/facilities.kml', options));
+dataSourcePromise.then(function(dataSource) {
+    var pixelRange = 15;
+    var minimumClusterSize = 3;
+
+    dataSource.clustering.enabled = true;
+    dataSource.clustering.pixelRange = pixelRange;
+    dataSource.clustering.minimumClusterSize = minimumClusterSize;
+
+    var viewModel = {
+        pixelRange: pixelRange,
+        minimumClusterSize: minimumClusterSize
+    };
+    Cesium.knockout.track(viewModel);
+
+    var toolbar = document.getElementById('toolbar');
+    Cesium.knockout.applyBindings(viewModel, toolbar);
+
+    function subscribeParameter(name) {
+        Cesium.knockout.getObservable(viewModel, name).subscribe(
+                function(newValue) {
+                    dataSource.clustering[name] = newValue;
+                }
+        );
+    }
+
+    subscribeParameter('pixelRange');
+    subscribeParameter('minimumClusterSize');
+
+    Sandcastle.addToolbarButton('Toggle Clustering', function() {
+        dataSource.clustering.enabled = !dataSource.clustering.enabled;
+    }, 'toggleClustering');
+
+    var removeListener;
+
+    var pinBuilder = new Cesium.PinBuilder();
+    var pin20 = pinBuilder.fromText('20+', Cesium.Color.RED, 48).toDataURL();
```

Comment:
Given how many labels there are, I think this would show better with a few levels above "20+"